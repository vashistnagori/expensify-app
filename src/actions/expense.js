import uuid from 'uuid';

export const addExpense=({description='', note='', amount=0, createdAt=''}={})=>{
    return {
        type:"addExpense", 
        expenseDetail:{id:uuid(), description, note, amount, createdAt}
    }
    }
    
export const removeExpense=({id=0}={})=>{
    return {
        type:"removeExpense", 
        id
    }
    }
    
export const editExpense=(id, updatedExpense)=>{
        return {
            type:"editExpense", 
            id,
            updatedExpense
        }
    }
