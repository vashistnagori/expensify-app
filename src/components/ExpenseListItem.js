import React from 'react';
import {Link} from 'react-router-dom';

const ExpenseListItem= ({id, description, amount, createdAt})=>{
    const to="/edit/"+id;
return (
    <div>
    <Link to={to}> <h3>{description}</h3></Link>
   <p>{amount}  - {createdAt}</p> 
   <hr/>
    </div>
)}
;

export default ExpenseListItem;
