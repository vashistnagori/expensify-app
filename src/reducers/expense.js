
export const expense_reducer=(ps=[],action)=>{
    switch(action.type){
    case("addExpense"):
    return [...ps, action.expenseDetail];
    break;
    case("editExpense"):
    let updatedExpenseArray=ps.map((current)=>{
        if(current.id === action.id){
         return {
             ...current,
             ...action.updatedExpense
         }
        }
        else{
          return current;
        }
    });
    return updatedExpenseArray;
    break;
    case("removeExpense"):
    return ps.filter(o => o.id !== action.id);;
    break;
    case("SET_EXPENSE"):
    return action.expense;
    break;
    default:
    return ps;
    }
    };