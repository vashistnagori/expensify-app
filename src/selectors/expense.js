import moment from 'moment';

export const filterExpense=(exp, { text, SortBy, startDate, endDate})=>{
    const filtered_expences=exp.filter((c)=>{
        const createdAtMoment= moment(c.createdAt);
        const match_start_date= startDate ? startDate.isSameOrBefore(createdAtMoment, "day") : true ;
        const match_end_date= endDate ? endDate.isSameOrAfter(createdAtMoment, "day") : true;
        const match_text= text==="" || c.description.toLowerCase().indexOf(text.toLowerCase()) > -1;
    return match_text && match_start_date && match_end_date;
    }).sort((a,b)=>{
    if(SortBy=='date'){
        return a.createdAt>=b.createdAt? -1 : 1;
    }else if(SortBy=='amount'){
        return a.amount>=b.amount? -1: 1;
    }
    });
    return filtered_expences;
    }