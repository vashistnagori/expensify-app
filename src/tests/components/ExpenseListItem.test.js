import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import {testexp} from '../fixture/expense';

test("testing ExpenseListItem with one expense data",()=>{
   const wrapper=shallow(<ExpenseListItem {...testexp[3]} />);
   expect(wrapper).toMatchSnapshot();
});