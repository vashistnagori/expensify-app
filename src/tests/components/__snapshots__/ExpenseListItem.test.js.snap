// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ExpenseListItem with one expense data 1`] = `
<div>
  <Link
    to="/edit/4"
  >
     
    <h3>
      mobile
    </h3>
  </Link>
  <p>
    110
      - 
    2419200000
  </p>
  <hr />
</div>
`;
